drivers/riello_usb.c

```diff
@@ -1079,14 +1082,33 @@ void upsdrv_updateinfo(void)
 	dstate_setinfo("input.bypass.frequency", "%.2f", DevData.Fbypass/10.0);
 	dstate_setinfo("output.frequency", "%.2f", DevData.Fout/10.0);
 	dstate_setinfo("battery.voltage", "%.1f", DevData.Ubat/10.0);
-	if ((DevData.BatCap < 0xFFFF) &&  (DevData.BatTime < 0xFFFF)) {
+        if (testvar("localcalculation")) {
```

Comment:
Looking at this code block some more, I wonder if it should be inverted in fact:
* if "native" readings make sense, use them; else
* if new flag is set, provide estimations; else
* print the warning (once)

although maybe not - if user had set the flag, we might have to obey always anyway

WDYT?

Reply:
It doesn't matter for users who reads the log files and manuals.
There are users that just USE their Open Media Vaults or DiskStations, and there they see only parameters returned by drivers. They don't see debug information - so maybe for them it would be good solution.
BTW. I've received reply from Riello.co.uk and they informed me that:

_The **idialog**, **iplug** and **Netpower** do not monitor certain parameters (It is not possible based on the design/architecture) therefore you will receive incorrect data._

So now we can exactly address the note to users of certain devices.